### PR TITLE
Added AppendLine method to widget.Entry

### PR DESCRIPTION
**Pull request by mbaklor**
### Description:

Added an `AppendLine` method to `entry`, to make appending to multiline entry easier.
Wasn't sure what to do if the entry isn't multiline so I set it to append with a space instead of a newline. Come to think of it maybe I should make the method just `Append`?

### Checklist:

- [x] Tests included.
- [x] Lint and formatter run with no errors.
- [x] Tests all pass.

#### Where applicable:

- [x] Public APIs match existing style and have Since: line.
- [ ] Any breaking changes have a deprecation path or have been discussed.
- [ ] Updated the vendor folder (using `go mod vendor`).
- [ ] Check for binary size increases when importing new modules.

### CODE REVIEW

**Comment by Jacalz** [comment]
Thanks for the contribution but I’m not so sure about this change. Appending text can be done very easily by the developer and I don’t think we should hard code it as lines. If you want to add a new line, you can just add a new line character. Adding with a space in between if it isn’t multiline also seems to be kind of unexpected. If you remove both of those parts of the methods it seems like something that can be done in one line by any developer wanting to append text.

**Comment by andydotxyz** [comment]
This is a tough one. I broadly agree with @Jacalz except that maybe this can be done more efficiently. The text provider has smarter ways to append text I think, so it could be called through to avoid the string manipulation route.

Regarding naming `Append` works if there are no more characters inserted (as this is not a "line based" editor). So maybe Append is one method and AppendLine a useful helper that only works on multi line?

File: widget/entry.go [hunk 1]
```diff
@@ -474,6 +474,19 @@ func (e *Entry) SetText(text string) {
 	e.updateCursorAndSelection()
 }
 
+// Appends the text to the end of the entry
```
Comment (andydotxyz) [inline widget/entry.go:477]: Sorry, but the comment should begin "Append" for the linter to be happy - i.e. "Append will add the given text to the end of the Entry"...

**Review by andydotxyz — APPROVED** [review]

**Comment by andydotxyz** [comment]
Thank so much for this!

**Pull request merged**